### PR TITLE
Subscriptions

### PR DESCRIPTION
**Pull request by helfer**
Just so I can leave comments 🙂

### CODE REVIEW

File: src/execution/execute.js [hunk 1]
```diff
@@ -120,22 +120,6 @@ export function execute(
   variableValues?: ?{[key: string]: mixed},
   operationName?: ?string
 ): Promise<ExecutionResult> {
-  invariant(schema, 'Must provide schema');
```
Comment (helfer) [inline src/execution/execute.js]: Why did you move the checks? I don't feel like I have the necessary context here.

File: .eslintrc [hunk 2]
```diff
@@ -70,7 +70,7 @@
     "eqeqeq": ["error", "smart"],
     "func-names": 0,
     "func-style": 0,
-    "generator-star-spacing": [2, {"before": true, "after": false}],
```
Comment (helfer) [inline .eslintrc]: Why change this rule?

File: src/execution/execute.js [hunk 3]
```diff
@@ -581,60 +584,68 @@ function resolveField(
...
   const resolveFn = fieldDef.resolve || defaultFieldResolver;
 
-  // The resolve function's optional third argument is a context value that
-  // is provided to every resolve function within an execution. It is commonly
-  // used to represent an authenticated user, or request-specific caches.
-  const context = exeContext.contextValue;
-
-  // The resolve function's optional fourth argument is a collection of
-  // information about the current execution state.
-  const info: GraphQLResolveInfo = {
```
Comment (helfer) [inline src/execution/execute.js]: Where is this type defined now?

File: src/subscription/subscribe.js [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+    exeContext,
+    type,
+    exeContext.operation.selectionSet,
+    Object.create(null),
+    Object.create(null)
+  );
+  const responseNames = Object.keys(fields);
+  invariant(
+    responseNames.length === 1,
+    'A subscription must contain exactly one field.'
```
Comment (helfer) [inline src/subscription/subscribe.js]: Should this say one "rootField"?

File: src/subscription/mapAsyncIterator.js [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+      }
+      return Promise.resolve({ value: undefined, done: true });
+    },
+    throw(error) {
+      if (typeof iterator.throw === 'function') {
+        return iterator.throw(error).then(mapResult);
+      }
+      return Promise.reject(error);
+    },
+    [$$asyncIterator]() {
```
Comment (helfer) [inline src/subscription/mapAsyncIterator.js]: What does this do?

**Comment by Urigo** [comment]
closing to open on the main graphql.js repo
@helfer can you please move your comments there?
